lib/network/modules/Clustering.js

```diff
@@ -696,6 +693,71 @@ class ClusterEngine {
     return stack;
   }
 
+  /**
+  * Using a clustered nodeId, update with the new options
+  * @param nodeId
+  * @returns {Array}
+  */
+  updateClusteredNode(nodeId, newOptions) {
+    var node = this.body.nodes[nodeId];
+    if (node !== undefined) {
+      node.setOptions(newOptions);
+      this.body.emitter.emit('_dataChanged');
+    }
+  }
+
+  /**
+  * Using a base edgeId, update all related clustered edges with the new options
+  * @param nodeId
+  * @returns {Array}
+  */
+  updateEdge(startEdgeId, newOptions) {
+    let allEdgeIds = this.getClusteredEdges(startEdgeId);
+    for (let i = 0; i < allEdgeIds.length; i++) {
+      var edge = this.body.edges[allEdgeIds[i]];
+      edge.setOptions(newOptions);      
+    }
+    this.body.emitter.emit('_dataChanged');
+  }
+
+  /**
+  * Get a stack of clusterEdgeId's (+base edgeid) that a base edge is the same as. cluster edge C -> cluster edge B -> cluster edge A -> base edge(edgeId)
+  * @param edgeId
+  * @returns {Array}
+  */
+  getClusteredEdges(edgeId) {
+    let stack = [];
+    let max = 100;
+    let counter = 0;
+
+    while (edgeId !== undefined && this.body.edges[edgeId] !== undefined && counter < max) {
```

Comment:
can you elaborate further?
this loop uses the new `edgeReplacedById` to move through the edges, a bit like a linked list.